utilities/write_batch_with_index/write_batch_with_index_internal.h

```diff
@@ -233,13 +256,24 @@ class WBWIIteratorImpl : public WBWIIterator {
   }
 
   void Seek(const Slice& key) override {
-    WriteBatchIndexEntry search_entry(&key, column_family_id_,
-                                      true /* is_forward_direction */,
-                                      false /* is_seek_to_first */);
+    WriteBatchIndexEntry search_entry(
+        beforeLowerBound(&key) ? iterate_lower_bound_ : &key, column_family_id_,
+        true /* is_forward_direction */, false /* is_seek_to_first */);
     skip_list_iter_.Seek(&search_entry);
   }
 
   void SeekForPrev(const Slice& key) override {
+    if (afterUpperBound(&key)) {
+      WriteBatchIndexEntry search_entry(
+          iterate_upper_bound_, column_family_id_,
+          // is_forward_direction, set to true so that
+          // iterate_upper_bound_ is excluded, this is not a readable trick,
+          // TODO fix it by renaming WriteBatchIndexEntry::is_forward_direction
+          true, false /* is_seek_to_first */);
+      skip_list_iter_.SeekForPrev(&search_entry);
```

Comment:
It seems the iterator can incorrectly return a key equal to `iterate_upper_bound_`. Maybe we can just do SeekToLast()?

Reply:
Do you think this can happen?

Reply:
From https://github.com/facebook/rocksdb/blob/d98a9cfb27dbffb4f7a7ab173b366dd626c2f721/db/write_batch.cc#L11-L12, it seems entries won't have offset = 0. However, I feel it's safer not to assume this, and just call SeekToLast().